### PR TITLE
Updated OpenAPI definitions for new endpoints

### PR DESCRIPTION
**Pull request by ruivieira**

### CODE REVIEW

File: docs/openapi.yaml [hunk 1]
```diff
@@ -79,10 +79,15 @@ definitions:
...
           type: string
+      type:
+        description: |-
+          Request type metadata
+        type: string
     required:
       - user
       - products
-  PredictionResponse:
+      - type
```
Comment (elmiko) [inline docs/openapi.yaml]: do we need to know the type?

we have a specific endpoint and payload for this type of request, seems unnecessary.

File: docs/openapi.yaml [hunk 2]
```diff
@@ -115,11 +120,92 @@ definitions:
...
+      user:
+        description: |-
+          The user identifier for the prediction
+        type: string
+      topk:
+        description: |-
+          The number of top ratings required
+        type: integer
+        minimum: 1
+      products:
```
Comment (elmiko) [inline docs/openapi.yaml]: do we even need products here since we are just asking for a top-k ranking?

File: docs/openapi.yaml [hunk 3]
```diff
@@ -115,11 +120,92 @@ definitions:
...
+        description: |-
+          An array of product identifiers for prediction
+        type: array
+        minItems: 1
+        items:
+          type: string
+      type:
```
Comment (elmiko) [inline docs/openapi.yaml]: same as for the ratings request, do we need the type here?

File: docs/openapi.yaml [hunk 4]
```diff
@@ -115,11 +120,92 @@ definitions:
...
+          type:
+            description: |-
+              Request type metadata
+            type: string
         required:
           - user
           - id
           - products
-
+          - type
```
Comment (elmiko) [inline docs/openapi.yaml]: probably don't need the type

File: docs/openapi.yaml [hunk 5]
```diff
@@ -115,11 +120,92 @@ definitions:
...
+                  type: string
+                rating:
+                  description: |-
+                    The predicted rating for this product
+                  type: number
+                  format: float
+              required:
+                - id
+                - rating
+          type:
```
Comment (elmiko) [inline docs/openapi.yaml]: similar as above, is type needed?

**Comment by ruivieira** [comment]
@elmiko thanks. OpenAPI and server responses updated as per review. Request type detection changed.

**Pull request merged**